Adding Standard Deduction

kumar-ab opened the pull request:
I have added the logic of Standard Deduction to the code. Also added an app for PIT results (appPIT.py) and the reforms of the 2019 budget.

sebastiansajie commented:
Looks Good. I shall send an app that incorporates the distribution tables structure that Martin has incorporated.

shekharkhetan commented:
@kumar-ab the logic of std. deduction is not so clear as to why you are adding 30000/- to everyone's income . Can you please clarify? Thanks

kumar-ab commented:
> @kumar-ab the logic of std. deduction is not so clear as to why you are adding 30000/- to everyone's income . Can you please clarify? Thanks

Standard Deduction has been introduced after removing the deduction related to two other allowances (i.e Medical allowance upto a maximum of Rs 15000 for the year and Transport allowance upto a maximum of Rs 1600 per month, which add up to maximum amount of deduction of Rs 34200).
The Salary income that is available in the ITR form is arrived after subtracting the Transport and Medical allowances. This deduction happens at the level of Form 16 (a form generated by the employer which is not captured in the ITR). There is no way to capture the actual deduction allowed to each of the tax payers under these heads. So, I estimated the average deduction under these two heads claimed by the tax-payers and added back the same in order to arrive at the salary on which the Standard Deduction has to be applied.
The estimate is based on the percentage of private sector (~ 80%) and government sector (~20%) salaried employees. The crude estimate gave a figure close to Rs 30000. So I added this figure and set the value of the policy variable "std_deduction" as 30000 for 2017 and 2018 to ensure that for these years the calculation doesn't get impacted. For the years 2019 onward the std_deduction can be set as per the reform.
